src/sentry/utils/cursors.py

```diff
@@ -1,92 +1,121 @@
+from __future__ import annotations
+
 from collections.abc import Sequence
+from typing import Any, Callable, Iterator, Protocol, Tuple, TypeVar, Union
+
+from sentry.utils.json import JSONData
+
+T = TypeVar("T")
+CursorValue = Union[float, int, str]
```

Comment:
A union isn't ideal here, but I couldn't manager to get anything better to work. I tried `TypeVar("CursorValue", float, int, str)` but ended up with error messages that I couldn't resolve in a reasonable amount of time. Could potentially revisit this later.